util/net/http_transport_mac.mm

```diff
@@ -289,6 +289,9 @@ static bool ExecuteProxyRequest(NSMutableURLRequest* request,
       @"HTTPSEnable" : @YES,
       @"HTTPSPort" : proxy_port,
       @"HTTPSProxy" : hostNS,
+      (__bridge id)kCFNetworkProxiesSOCKSEnable : @YES,
+      (__bridge id)kCFNetworkProxiesSOCKSPort : proxy_port,
+      (__bridge id)kCFNetworkProxiesSOCKSProxy : hostNS,
```

Comment:
these options are not available on iOS, so we should check our platform before trying to add them